src/server/endpoints/covidcast_meta.py

```diff
@@ -26,57 +26,66 @@ def __str__(self):
         get_structured_logger('server_api').warning("no data in covidcast_meta cache")
-        return
-
-    age = row["age"]
-    if age > max_age and row["epidata"]:
-        get_structured_logger('server_api').warning("covidcast_meta cache is stale", cache_age=age)
-
-    epidata = loads(row["epidata"])
-
-    if not epidata:
-        return
+        return printer(_nonerator())
 
-    def filter_row(row: Dict):
-        if time_types and row.get("time_type") not in time_types:
-            return False
-        if geo_types and row.get("geo_type") not in geo_types:
-            return False
-        if not signals:
-            return True
-        for signal in signals:
-            # match source and (signal or no signal or signal = *)
-            if row.get("data_source") == signal.source and (
-                signal.signal == "*" or signal.signal == row.get("signal")
-            ):
-                return True
-        return False
+    metadata_list = loads(metadata["epidata"])
 
-    for row in epidata:
-        if filter_row(row):
-            yield row
+    if not metadata_list:
+        # the db table has a row, but there is no metadata about any signals in it
+        get_structured_logger('server_api').warning("empty entry in covidcast_meta cache")
+        return printer(_nonerator())
 
+    standard_age = 60 * 60 # expected metadata regeneration interval, in seconds (currently 60 mins)
+    # TODO: get this ^ from a config var?  ideally, it should be set to the time between runs of
+    #       src/acquisition/covidcast/covidcast_meta_cache_updater.py
+    age = metadata["age"]
+    if age > standard_age * 1.25:
+        # complain if the cache is too old (currently, more than 75 mins old)
+        get_structured_logger('server_api').warning("covidcast_meta cache is stale", cache_age=age)
 
-@bp.route("/", methods=("GET", "POST"))
-def handle():
-    time_types = extract_strings("time_types")
-    signals = [SourceSignal(v) for v in (extract_strings("signals") or [])]
-    geo_types = extract_strings("geo_types")
-
-    return create_printer(request.values.get("format"))(filter_fields(fetch_data(time_types, geo_types, signals)))
+    def cache_entry_gen():
+        for entry in metadata_list:
+            if time_types and entry.get("time_type") not in time_types:
+                continue
+            if geo_types and entry.get("geo_type") not in geo_types:
+                continue
+            if not signals:
+                yield entry
+            for signal in signals:
+                # match source and (signal or no signal or signal = *)
+                if entry.get("data_source") == signal.source and (
+                    signal.signal == "*" or signal.signal == entry.get("signal")
+                ):
+                    yield entry
+
+    return printer(
+        filter_fields(cache_entry_gen()),
+        headers={
+            "Cache-Control": f"max-age={standard_age}, public",
+            "Age": f"{age}",
```

Comment:
i disagree.  i think it will in fact work, especially because (as i mentioned above) the R lib doesnt even seem to respect the "Age" header.

and we do want this -- when the time elapsed since the resource was cached plus `Age` exceeds `max-age`, there is very likely to be a newly calculated metadata waiting to be retrieved (presuming `max-age` is set to the metadata regeneration interval).

alternatively, we could use `"Cache-Control": f"max-age={max(standard_age-age, 1)}, public"` to achieve a similar effect without using the "Age" header.

Reply:
> as i mentioned above

can you link me to this? i can't seem to find it.

Reply:
use the link from [the opening comment](https://github.com/cmu-delphi/delphi-epidata/pull/1222#issue-1779832451) ( https://github.com/r-lib/httr/blob/HEAD/R/cache.R ) and search for "age"